pkg/server/router.go

```diff
@@ -38,8 +40,14 @@ func router(serverConfig *config.Control, tunnel http.Handler, cacertsGetter CAC
 	authed.Use(authMiddleware(serverConfig))
 	authed.NotFoundHandler = serverConfig.Runtime.Handler
 	authed.Path("/v1-k3s/connect").Handler(tunnel)
-	authed.Path("/v1-k3s/node.crt").Handler(nodeCrt(serverConfig))
-	authed.Path("/v1-k3s/node.key").Handler(nodeKey(serverConfig))
+	authed.Path("/v1-k3s/serving-kube-apiserver.crt").Handler(servingKubeAPICert(serverConfig))
+	authed.Path("/v1-k3s/serving-kube-apiserver.key").Handler(servingKubeAPIKey(serverConfig))
```

Comment:
does this key still need to be exported this way?

Reply:
The node.(crt/key) was a combo server/client cert that has been split up into individual certs, serving-kube-apiserver is a serving cert used for the kubelet tls-cert-file/tls-private-key-file args.

Reply:
This has been changed to a specially generated `serving-kubelet.key` instead of re-using the `serving-kube-apiserver.key`. The related cert has been updated to use SANs only for that node. It appears we still need to create a server signed cert for things like metrics-server to function correctly. Thanks for looking at this @yamt, I think the PR is almost ready, still a little cleanup and hopefully future changes to simplify stuff but if you have any other comments they are greatly appreciated! :)